packages/main/src/RPA/MFA.py

```diff
@@ -0,0 +1,142 @@
+from enum import Enum
+import logging
+from typing import Optional
+from pyotp import HOTP, TOTP
+from robot.api.deco import keyword
+from RPA.Robocorp.Vault import Vault
+
+
+class OTPMode(Enum):
+    """Enumeration for type of TOP to use"""
+
+    TIME = "TIME"
+    COUNTER = "COUNTER"
+
+
+class TOTPNotSetError(Exception):
+    "Error when TOTP (Time-based One-Time Password) has not been set"
+
+
+class HOTPNotSetError(Exception):
+    "Error when HOTP (HMAC One-Time Password) has not been set"
+
+
+TOTP_NOT_SET_ERROR_MSG = """TOTP (Time-based One-Time Password) can be set in library initialization, with
+`Use MFA Secret From Vault` keyword or with `Set Time Based OTP` keyword."""
+
+HOTP_NOT_SET_ERROR_MSG = """HOTP (HMAC One-Time Password) can be set in library initialization, with
+`Use MFA Secret From Vault` keyword or with `Set Counter Based OTP` keyword."""
+
+
+class MFA:
+    """*RPA.MFA* is a library for generating one-time passwords (OTP).
+
+    Based on the `pyotp <https://pypi.org/project/pyotp/>`_ package.
+
+
```

Comment:
Do we need some explanation here on what is expected from the Vault secret content?

Reply:
Yes the instructions on the **secret** - like how to get it, will be included in the example. Need to take some screenshots and build up on the examples / articles.